reporting/reporting.go

```diff
@@ -0,0 +1,137 @@
+package reporting
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/honeycombio/dynsampler-go"
+	"github.com/honeycombio/libhoney-go"
+
+	"github.com/honeycombio/honeytail/event"
+)
+
+const (
+	contextKeyBuilder = "builder"
+	contextKeySampler = "sampler"
+	reportingSuffix   = "-reports"
+)
+
+func NewContext(ctx context.Context) context.Context {
+	builder := libhoney.NewBuilder()
+	builder.Dataset += reportingSuffix
+	sampler := &dynsampler.PerKeyThroughput{
+		ClearFrequencySec:      1,
+		PerKeyThroughputPerSec: 1000,
```

Comment:
Psst, a sanity check on these values would be great.

I wish I could configure dynsampler to check after the first second and then adjust every 30, but that seems... not in scope here.

Reply:
Per key throughput of 1000 is *super* high.